### PR TITLE
Changes to improve vegetation health at high latitudes

### LINKED ISSUE #2298: Changes to xdrdt in the parameter file have no effect because of a hard-coded flag set to .false. (flg_snoage_scl    = .false.)
Changes to xdrdt in the parameter file have no effect because of a hard-coded flag set to .false. in SnowSnicarMod.F90

  ! scaling of the snow aging rate (tuning option):
  logical :: flg_snoage_scl    = .false.                 ! flag for scaling the snow aging rate by some arbitrary factor

            ! Multiply rate of change of effective radius by some constant, xdrdt
            if (flg_snoage_scl) then
               dr = dr*params_inst%xdrdt
            endif

The suggested fix is to remove flg_snoage_scl completely such that xdrdt will always have an effect if changed in the parameter file.

### PR DESCRIPTION
**Pull request by olyson**
### Description of changes

Goal is to improve vegetation health/survivability at high latitudes (address LMWG_dev discussion [#3](https://github.com/NCAR/LMWG_dev/discussions/3))

### Specific notes

These changes include:

Remove snicar_snobc_intmix from EXPERIMENTAL endrun (allow it to be true)
Remove flg_snoage_scl in SNICAR such that xdrdt can have an effect (fixes #2298 )
New parameter file and namelist values for clm5_1:

1. snow_thermal_cond_method = Sturm1997 (default for clm5_1)
2. snicar_snobc_intmix = .true. (default for clm5_1)
3. ctsm51_params.c240208.nc is the new CTSM parameter file (changes: froot_leaf(11:12)=1.2, FUN_fracfixers(11:12)=1, xdrdt=5, scvng_fct_mlt_sf=0.5, snw_rds_refrz=1500, fresh_snw_rds_max=400)
4. New history fields for coupler history verification (default off)
5. Add snow5d_thresh_for_onset to parameter file, set to 0.2 for clm51 and 0.1 (unchanged) for clm50 and clm45.

Contributors other than yourself, if any: @wwieder , @dlawrenncar , @slevis-lmwg , LMWG members

CTSM Issues Fixed (include github issue #): LMWG_dev discussion [#3](https://github.com/NCAR/LMWG_dev/discussions/3) and #2298 

Are answers expected to change (and if so in what way)? Yes, new climate

Any User Interface Changes (namelist or namelist defaults changes)? Yes, new namelist defaults including new parameter files:
ctsm51_params.c240208.nc
clm50_params.c240208.nc
clm45_params.c240208.nc
ctsm51_ciso_cwd_hr_params.c240208.nc

Testing performed, if any:
Replicate simulation in LMWG_dev issue [#51](https://github.com/NCAR/LMWG_dev/issues/51) as bfb when snow5d_thresh_for_onset on ctsm51_ciso_cwd_hr_params.c240208.nc was set to original value of 0.1.

aux_clm on Derecho:
No unexpected failures.
NLCOMP failures as expected.
clm51 compsets fail baseline comparison as expected.
clm50 and clm45 pass baseline comparisons as expected.
These EXPECTED FAILURE tests passed:
    PASS SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.derecho_gnu.clm-default--clm-NEON-NIWO RUN time=62 (UNEXPECTED: expected FAIL)
    PASS SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Bgc.derecho_gnu.clm-NEON-MOAB--clm-PRISM RUN time=79 (UNEXPECTED: expected FAIL)
    PASS SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Fates.derecho_gnu.clm-FatesFireLightningPopDens--clm-NEON-FATES-NIWO RUN time=52 (UNEXPECTED: expected FAIL)
    PASS SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51Fates.derecho_gnu.clm-FatesPRISM--clm-NEON-FATES-YELL RUN time=35 (UNEXPECTED: expected FAIL)
    PASS SMS_Ld10_D_Mmpi-serial.CLM_USRDAT.I1PtClm51SpRs.derecho_gnu.clm-default--clm-NEON-TOOL RUN time=33 (UNEXPECTED: expected FAIL)

### CODE REVIEW

**Review by ekluzek — CHANGES_REQUESTED** [review]
@olyson this is really nice to see become a PR. It'll be a nice update.

I have a couple things that we need to do here. You won't necessarily be the one that does them, but I marked them as something that needs to happen.

It also looks to me like this will also change answers for clm45 and clm50 physics, and I wasn't sure if that's acceptable or not? So there's some questions about that...

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml:489]: Right, that will need to be done by someone with permissions (I don't have them).

File: src/biogeophys/SnowSnicarMod.F90
Comment (olyson) [reply src/biogeophys/SnowSnicarMod.F90:1659]: I haven't done aux_clm testing for this yet, but I haven't changed the clm45 and clm50 parameter files, which contain xdrdt=1, so my expectation is that this wouldn't change answers.

File: src/main/controlMod.F90
Comment (olyson) [reply src/main/controlMod.F90:632]: Good catch, thanks.

File: src/biogeophys/SnowSnicarMod.F90
Comment (ekluzek) [reply src/biogeophys/SnowSnicarMod.F90:1659]: Ahh, perfect. Yeah it shouldn't then. And at most it would be roundoff different, which would certainly be fine...

Comment (olyson) [reply src/biogeophys/SnowSnicarMod.F90:1659]: I believe I verified it as bfb in the PPE testing I did, but will check again when I do the testing.

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (slevis-lmwg) [reply bld/namelist_files/namelist_defaults_ctsm.xml:489]: I'm happy to do it. Please remind me when you're ready.

File: src/main/controlMod.F90
Comment (olyson) [reply src/main/controlMod.F90:632]: Done

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml]: I've now created new parameters files for clm51, clm50, and clm45 that have snow5d_thresh_for_onset on them:
ctsm51_params.c240208.nc
clm50_params.c240208.nc
clm45_params.c240208.nc
ctsm51_ciso_cwd_hr_params.c240208.nc

clm50 and clm45 aux_clm tests are bfb because snow5d_thresh_for_onset is set to its original value.
NOTE that I started from the parameter files that @samsrabin generated yesterday (PR #2350 ).

File: src/main/controlMod.F90
Comment (olyson) [reply src/main/controlMod.F90:632]: I've rework the logic in CLMBuiildNamelist.pm because my original change wasn't working correctly.
Individually, now snicar_snobc_intmix = .false.  = .true. or .false. and snicar_snodst_intmix = .false.   We check for snicar_snodst_intmix = .true. and flag it as UNSUPPORTED in the first section of code.

The possible combinations of snicar_snobc_intmix and snicar_snodst_intmix are:

(1) snicar_snobc_intmix = .false.    snicar_snodst_intmix = .false.
(2) snicar_snobc_intmix = .true.    snicar_snodst_intmix = .false.
(3) snicar_snobc_intmix = .false.    snicar_snodst_intmix = .true.
(4) snicar_snobc_intmix = .true.    snicar_snodst_intmix = .true.

In the second section of code, (4) is flagged as "do not work together".  (1), (2), and (3) are not flagged, however, (3) is flagged as UNSUPPORTED in the first section of code because snicar_snodst_intmix is not allowed to be .true.

Tested using the various combinations in clm_short tests.
I was trying to preserve the original code as much as possible, but there may be a better way to check the valid combinations, e.g., seems like one could simply check for both .true. and flag it.

Comment (ekluzek) [reply src/main/controlMod.F90:632]: Oh @olyson just remove snicar_snobc_intmix from the list. There's separate checking to make sure it's a valid FORTRAN logical.

So you just remove that one line, and you are good.

Comment (olyson) [reply src/main/controlMod.F90:632]: I had removed that line but the code after that was flagging the valid .false., .false. case as invalid.

File: src/main/atm2lndType.F90
Comment (wwieder) [inline src/main/atm2lndType.F90:1]: Just wondering, why are all these changes needed?

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (wwieder) [inline bld/namelist_files/namelist_defaults_ctsm.xml:444]: should we be specific about different physics versions (4.5, 5.0, 5.1) and avoid setting a default?  My only thought here is to make sure we identify the default when we start working on the > 5.1 series? Happy to defer to convention used elsewhere in the namelist files

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (wwieder) [inline bld/namelist_files/namelist_defaults_ctsm.xml:1629]: as above, with defining particular physics versions, instead of a default.

File: src/biogeophys/SnowSnicarMod.F90
Comment (wwieder) [inline src/biogeophys/SnowSnicarMod.F90:1]: were all the other snicar chagnes we modified on the parameter file, @olyson ?  I can't remember.

**Review by wwieder — COMMENTED** [review]
looks good, @olyson just a few comments / questions.

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml]: I redid the parameter files so that they are based on @samsrabin 's latest files (c240207b) and repeated the testing.  Same (good) results as reported previously.

File: src/main/atm2lndType.F90
Comment (olyson) [reply src/main/atm2lndType.F90:1]: These are to produce history fields that are suitable for validating CPLHIST mode, as we discussed in https://github.com/ESCOMP/CTSM/discussions/1951.  They are off by default.  I just didn't want to have to re-create that work the next we need to validate CPLHIST

File: src/biogeophys/SnowSnicarMod.F90
Comment (olyson) [reply src/biogeophys/SnowSnicarMod.F90:1]: Yes: xdrdt=5, scvng_fct_mlt_sf=0.5, snw_rds_refrz=1500, fresh_snw_rds_max=400

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml:444]: I'm not sure what line(s) you are pointing to.  Are you talking about these lines?

`<snow_thermal_cond_method>Jordan1991</snow_thermal_cond_method>`
`<snow_thermal_cond_method phys="clm5_1">Sturm1997</snow_thermal_cond_method>`

I was following the recent example for Meier:

`<z0param_method>ZengWang2007</z0param_method>`
`<z0param_method phys="clm5_1">Meier2022</z0param_method>`

Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml:444]: I guess your point is, should we be doing something like this:

`<snow_thermal_cond_method phys="clm4_5>Jordan1991</snow_thermal_cond_method>`
`<snow_thermal_cond_method phys="clm5_0>Jordan1991</snow_thermal_cond_method>`
`<snow_thermal_cond_method phys="clm5_1">Sturm1997</snow_thermal_cond_method>`

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (olyson) [reply bld/namelist_files/namelist_defaults_ctsm.xml:1629]: Again, this follows the Meier example.

Comment (wwieder) [reply bld/namelist_files/namelist_defaults_ctsm.xml:1629]: That's fine with me if it's consistent with how we're defining defaults elsewhere?

**Comment by ekluzek** [comment]
Just FYI on the latest I heard about the cesm2_3_beta17 tag. The next alpha tag is close to being done. And there are two tags after it. The first one has several CAM changes that need to come in that are likely a few weeks away. It would be really great to get this into cesm2_3_beta17, so hopefully that will give time for this to be figured out before then.

So the beta tag is roughly a few weeks to a month away...

**Comment by wwieder** [comment]
@olyson the scientific tests you ran look good.  Is it worth discussing this?  
Are we ready to bring this PR into main?

**Comment by olyson** [comment]
Maybe we can discuss at our meeting tomorrow.  I think it is probably as good as we can get it for now.  A question that came up in the LMWG meeting was whether to do some kind of elevation threshold for the Sturm approach, but that would require more research, etc.  And maybe that would be more important for high-resolution simulations.  We'll also have to think about what initial file we'll want to use in conjunction with these changes in the next coupled model set of simulations.

**Comment by wwieder** [comment]
I agree, the alpine vs. tundra distinction is something that needs more scientific evaluation as we as conversations about SE support.  I suggest we table this option for now.  Moreover, for regional or point runs users can still switch between schemes using the namelist options.

We should discuss the initial condition question, however.  Is a longer spinup needed to provide better initial conditions?  Maybe this would be a good idea anyway if the 5.2 PR is close to coming to main?  We can discuss tomorrow.

**Comment by ekluzek** [comment]
@slevis-lmwg cool! I moved it up in upcoming tags. @rgknox has the next tag for FATES history dimension control. And this should then be next.

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (ekluzek) [reply bld/namelist_files/namelist_defaults_ctsm.xml:444]: I agree with @olyson and @wwieder I think the default should be explicit for each clm physics version.

File: bld/namelist_files/namelist_defaults_ctsm.xml
Comment (ekluzek) [inline bld/namelist_files/namelist_defaults_ctsm.xml:489]: I think @slevis-lmwg already knows this, but the new files need to be put into CESM inputdata and rimported into svn.

File: cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm
Comment (ekluzek) [inline cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: Move this into cesm inputdata as talked about above.

**Review by ekluzek — APPROVED** [review]
I went through this again since it will be coming in right away. @slevis-lmwg already knows this, but I just pointed out about moving files to CESM inputdata.

There's some discussion about namelist settings and specifying the default for all clm physics versions. Technically if you do...

``` xml
<namelist_thing>blah</namelist_thing>
<namelist_thing phys="clm5_1">bigger blah</namelist_thing>
```

That's equivalent to

``` xml
<namelist_thing phys="clm4_5">blah</namelist_thing>
<namelist_thing phys="clm5_0">blah</namelist_thing>
<namelist_thing phys="clm5_1">bigger blah</namelist_thing>
```
BUT, the advantage of the later is that if you don't set ALL the clm physics versions -- it'll complain if that namelist setting is required. So it'll flag an error when we bring in a new physics version for the first time. I think it's also clearer.

File: cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm
Comment (slevis-lmwg) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: Moved and rimported. Now I need to change the paths accordingly in these .xml and user_nl_clm files.

Comment (wwieder) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: @olyson maybe your GSWP3 spinup in #2403 should point to this parameter set from inputdata?  Not sure it will be critical since they are the same file...?

Comment (olyson) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: The only difference between these files in the most recent version of CTSM:

ctsm51_params.c240207b.nc
ctsm51_ciso_cwd_hr_params.c240207b.nc

is "ceta" which is 450 in the ciso version and 358 in the normal version.  This parameter was changed from 450 to 358 in PR #250 .
I'm guessing that the ciso version didn't get changed at that point.
So I think I should be using the 358 version.  I think we decided we should get rid of the ciso version at some point.

Comment (olyson) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: Oh, maybe I misunderstood your question.  You are suggesting I point to the non-iso rimported inputdata file, maybe to make sure it was imported correctly and readable, etc.?

Comment (olyson) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: Turns out that was a good idea.  I can't read that file.  @slevis-lmwg can you change the permissions on /glade/campaign/cesm/cesmdata/inputdata/lnd/clm2/paramdata/ctsm51_params.c240208.nc?

Comment (slevis-lmwg) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: Ok, done.

Comment (olyson) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: I'm getting an error using this surface dataset with the deadveg branch (base is ctsm5.1.dev166):

/glade/campaign/cesm/cesmdata/inputdata/lnd/clm2/surfdata_esmf/ctsm5.2.0/surfdata_0.9x1.25_hist_1850_78pfts_c240216.nc

So maybe there IS a code modification that is not yet in the main code but is in the ctsm5.2_mksurfdata branch?  The error message is:

surfrd_get_data ERROR: sum of wt_lunit not    1.00000000000000       at nl=4
sum is:   0.000000000000000E+000

Something to do with PCT_OCEAN maybe?

Comment (slevis-lmwg) [reply cime_config/testdefs/testmods_dirs/clm/ciso_cwd_hr/user_nl_clm:1]: You're probably right that it's the PCT_OCEAN in the new fsurdat file.

I recommend this pre-PCT_OCEAN fsurdat file, which I think should be the same in other respects:
surfdata_0.9x1.25_hist_1850_78pfts_c231026.nc

**Comment by slevis-lmwg** [comment]
I will hold off running the test-suites until I update to dev172.

**Comment by slevis-lmwg** [comment]
@olyson could you look over my ChangeLog entry and let me know what you think?

File: doc/ChangeLog
Comment (olyson) [inline doc/ChangeLog:18]: ctsm51_params.c240208.nc is the version.  You probably had 240201 because I had that in my draft PR information, which I've since changed.

**Review by olyson — COMMENTED** [review]
One comment regarding the parameter date, otherwise, it looks good to me.  I suppose that instead of saying "new climate" we could say "new climate at high latitudes".

File: doc/ChangeLog
Comment (slevis-lmwg) [reply doc/ChangeLog:18]: Thank you for catching that!

**Comment by slevis-lmwg** [comment]
@ekluzek helped me resolve the ./manage_externals error with the following (I will post this in the CTSM_SE chat, too):
```
cd /src/fates
git fetch --all
cd ../..
./manage_externals/checkout_externals
```
Now I have submitted the test-suites.

**Comment by olyson** [comment]
I was just looking at the test results on derecho for this which I assume is:

/glade/derecho/scratch/slevis/tests_0314-145540de

I noticed that there are differences with baseline for some Clm50 tests (all fates tests).  Based on my original testing, I wouldn't expect this.  My results showed no differences for all Clm50 tests.  Do we know why these baseline comparisons fail?

**Comment by slevis-lmwg** [comment]
Sorry for overlooking that. I do not know why...

**Comment by olyson** [comment]
Given the discussion I saw regarding fates externals, I wonder if it has something to do with that?

**Comment by slevis-lmwg** [comment]
I was about to say the same.

**Comment by slevis-lmwg** [comment]
I am open to suggestions on how to proceed @ekluzek

**Comment by olyson** [comment]
Just noting that there are a couple of Clm45Fates DIFF failures as well.

**Comment by ekluzek** [comment]
Here's the list of changed tests that I'm seeing (for non Clm51):

ERP_Ld9.f45_f45_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesColdAllVars
ERS_D_Ld3_PS.f09_g17.I2000Clm50FatesRs.derecho_intel.clm-FatesCold
ERS_D_Ld5.f10_f10_mg37.I2000Clm50Fates.derecho_intel.clm-FatesCold
ERS_D_Mmpi-serial_Ld5.1x1_brazil.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold
ERS_D_Mmpi-serial_Ld5.5x5_amazon.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold
ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesColdFixedBiogeo
ERS_Ld30.f45_f45_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesColdSizeAgeMort
ERS_Ld9.f10_f10_mg37.I2000Clm50FatesCruRsGs.derecho_intel.clm-FatesColdCH4Off
SMS_D_Ld5.f10_f10_mg37.I2000Clm45Fates.derecho_intel.clm-FatesCold
SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold
SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesCold
SMS_D_Lm6_P256x1.f45_f45_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesCold
SMS_Ld5.f10_f10_mg37.I2000Clm45Fates.derecho_intel.clm-FatesCold
SMS_Ld5.f10_f10_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesCold
SMS_Ld5_PS.f19_g17.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold

**Comment by rgknox** [comment]
I'm looking through this more closely, but I think that the change to "dr" should propogate into FATES results as well, which should explain the diffs. 

https://github.com/ESCOMP/CTSM/pull/2348/files#diff-a313a6093d335d65fdb56abaedbbc3416f8b8f5f095f828ddd703f907d57526bR1659

**Comment by slevis-lmwg** [comment]
> I'm looking through this more closely, but I think that the change to "dr" should propogate into FATES results as well, which should explain the diffs.
> 
> https://github.com/ESCOMP/CTSM/pull/2348/files#diff-a313a6093d335d65fdb56abaedbbc3416f8b8f5f095f828ddd703f907d57526bR1659

@olyson didn't think so according to this [conversation](https://github.com/ESCOMP/CTSM/pull/2348/files#r1480365177), but - just in case - I'm trying `SMS_Ld5_PS.f19_g17.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold` (the last test in Erik's list above) as follows:
```
git checkout ctsm5.1.dev173
git switch -c ctsm5.1.dev173
./manage_externals/checkout_externals
vi SnowSnicarMod.F90
```
I changed line 1659 to what's in dev174 (same as Ryan's link in the last post).

**Comment by slevis-lmwg** [comment]
As @olyson would expect, the last test returned:
`PASS SMS_Ld5_PS.f19_g17.I2000Clm50FatesRs.derecho_gnu.clm-FatesCold BASELINE /glade/campaign/cgd/tss/ctsm_baselines/ctsm5.1.dev174:`

**Comment by slevis-lmwg** [comment]
Is my test design wrong or does this end up showing that the specific line does change answers in Clm50? In the next 45' until we meet, I will try a few more tests.

**Comment by slevis-lmwg** [comment]
@rgknox will rerun the dev173 baselines.
Then I will rerun a clm45 + clm50 test on each machine (izumi + derecho) to confirm that things work.

**Comment by slevis-lmwg** [comment]
I reran the following tests and confirmed that Clm45 and Clm50 are bit-for-bit against dev173 now:
```
PASS SMS_Ld5.f10_f10_mg37.I2000Clm45Fates.derecho_intel.clm-FatesCold BASELINE ctsm5.1.dev173:
PASS SMS_Ld5.f10_f10_mg37.I2000Clm50FatesRs.derecho_intel.clm-FatesCold BASELINE ctsm5.1.dev173:
PASS SMS_D_Ld5.f10_f10_mg37.I2000Clm50FatesRs.izumi_nag.clm-FatesCold BASELINE ctsm5.1.dev173:
PASS SMS_Ld5.f10_f10_mg37.I2000Clm45Fates.izumi_intel.clm-FatesCold BASELINE ctsm5.1.dev173:
```

**Pull request merged**
